ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/X11/tasks/main.yml

```diff
@@ -1,46 +1,83 @@
 #########################################################################
 # Install X11 extensions                                                #
+# x11.base.lib installp download requiring an IBMid                     #
+# https://www.ibm.com/support/pages/fileset-information-x11baselib      #
 # x11.adt.ext installp download requiring an IBMid                      #
-# https://www-01.ibm.com/support/docview.wss?uid=isg1fileset-1198156818 #
+# https://www.ibm.com/support/pages/fileset-information-x11adtext       #
 # x11.vfb installp download requiring an IBMid                          #
-# http://www-01.ibm.com/support/docview.wss?uid=isg1fileset-2054486724  #
+# https://www.ibm.com/support/pages/fileset-information-x11vfb          #
 #########################################################################
 ---
-- name: Check if X11.adt.ext is installed
-  shell: lslpp -l X11.adt.ext >/dev/null 2>&1
-  register: does_X11_adt_ext_exist
-  ignore_errors: yes
+# Rather than run lslpp - use stat on the lpp directories
+# /usr/lpp/X11.base/X11.base.lib      X11.base.lib
+# /usr/lpp/X11/include/X11/extensions X11.adt.ext
+# /usr/lpp/X11.vfb                    X11.vfb
+- name: Verify X11 software
+  block:
+  - name: Verify X11 base library
+    stat:
+      path: /usr/lpp/X11.base/X11.base.lib
+    register: _base
+  - name: Verify X11 extensions requirements
+    stat:
+      path: /usr/lpp/X11/include/X11/extensions
+    register: _ext
+  - name: Verify X11 Frame Buffer Requirements
+    stat:
+      path: /usr/lpp/X11.vfb
+    register: _vfb
+  - name: Verify Readiness #1
+    set_fact:
+      _x11rdy: "{{ _base.stat.exists
+        and _ext.stat.exists
+        and _vfb.stat.exists }}"
   tags: x11
 
-- name: Check if X11.vfb is installed
-  shell: lslpp -l X11.vfb >/dev/null 2>&1
-  register: does_x11_exist
-  ignore_errors: yes
-  tags: x11
+- name: Install from Vendor Files
+  block:
+    - name: Unpack Vendor Files
+      unarchive:
+        src: /Vendor_Files/aix/OpenGL_X11.tar
+        dest: /tmp
+        remote_src: no
+      ignore_errors: yes
+      register: _vendor_copied
 
-- name: Transfer and extract X11
-  unarchive:
-    src: /Vendor_Files/aix/OpenGL_X11.tar
-    dest: /tmp
-    remote_src: no
-  when: does_x11_exist.rc != 0 or does_X11_adt_ext_exist.rc != 0
-  tags: x11
+    - name: Install IBM X11 Extensions - installp
+      command:
+        installp -e /tmp/x11.log -aXYgd /tmp/x11 X11.adt.ext X11.vfb X11.base.lib
```

Comment:
By only specifying /tmp/x11, the system isnt able to find installation media for X11.adt.ext

Ansible error log:
```
FAILURES\n
--------\n
  Filesets listed in this section failed pre-installation verification\n  and will not be installed.\n\n  Missing Filesets\n
  ----------------\n
  The following filesets could not be found on the installation media.\n
  If you feel these filesets really are on the media, check for typographical\n
  errors in the name specified or, if installing from directory, check for\n
  discrepancies between the Table of Contents file (.toc) and the images that\n
  reside in the directory.\n
\n

    X11.adt.ext                              \n
    X11.base.lib
```

And then from running the command directly on the machine

```
root@p9-aix1-ojdk06:[/root]installp -aXYgd x11 X11.adt.ext X11.vfb X11.base.lib
+-----------------------------------------------------------------------------+
                    Pre-installation Verification...
+-----------------------------------------------------------------------------+
Verifying selections...done
Verifying requisites...done
Results...

FAILURES
--------
  Filesets listed in this section failed pre-installation verification
  and will not be installed.

  Missing Filesets
  ----------------
  The following filesets could not be found on the installation media.
  If you feel these filesets really are on the media, check for typographical
  errors in the name specified or, if installing from directory, check for
  discrepancies between the Table of Contents file (.toc) and the images that
  reside in the directory.

    X11.adt.ext                              
    X11.base.lib                             

  << End of Failure Section >>

```

Only by running `installp -aXYgd /x11/X11.adt X11.adt.ext` could I get X11.adt.ext to install.

Also, I dont think the Vendor file for x11 has X11.base.lib on it

```
root@p9-aix1-ojdk06:[/root]ls -la x11
total 61432
drwxr-xr-x 2  501 staff       256 Oct 22  2019 .
drwx------ 6 root system     4096 Mar 18 15:32 ..
-rw-r--r-- 1  501 staff      5933 Aug 20  2019 .toc
-rw-r--r-- 1  501 staff  21670912 Apr 17  2018 OpenGL.OpenGL_X.dev
-rw-r--r-- 1  501 staff  35705856 Apr 17  2018 OpenGL.OpenGL_X.rte
-rw-r--r-- 1 root system  5476352 Oct 22  2019 X11.adt
-rw-r--r-- 1  501 staff     33792 Apr 17  2018 X11.vfb
```

hence why only vfb and adt.ext were installed with vendor file previously

Reply:
removed from installp command, but still check it is installed - or not.